cannon/mipsevm/program/load.go

```diff
@@ -42,12 +42,18 @@ func LoadELF[T mipsevm.FPVMState](f *elf.File, initState CreateInitialFPVMState[
 			}
 		}
 
-		// TODO(#12205)
-		if prog.Vaddr+prog.Memsz >= uint64(1<<32) {
-			return empty, fmt.Errorf("program %d out of 32-bit mem range: %x - %x (size: %x)", i, prog.Vaddr, prog.Vaddr+prog.Memsz, prog.Memsz)
+		if prog.Memsz == 0 {
+			// Nothing to do
+			continue
+		}
+
+		lastByteToWrite := prog.Vaddr + prog.Memsz - 1
+		lastMemoryAddr := uint64(^arch.Word(0))
```

Comment:
For cannon64, the virtual addressing should be limited to 48-bits. It's not a MIPS64 limitation, but rather an assumption many userspace programs rely on for tagged pointers and such.
So the last memory address value should be 2^48-1.